cleo_plugins/IniFiles/IniFiles.cpp

```diff
@@ -188,23 +144,17 @@ class IniFiles
 		0AF5=4,write_string %1s% to_ini_file %2s% section %3s% key %4s%
 		****************************************************************/
 	{
-		char path[MAX_PATH];
-		char sectionName[100];
-		char key[100];
-		char strValue[100];
-		BOOL result;
-
-		CLEO_ReadStringPointerOpcodeParam(thread, strValue, sizeof(strValue));
-		CLEO_ReadStringPointerOpcodeParam(thread, path, sizeof(path));
-		CLEO_ReadStringPointerOpcodeParam(thread, sectionName, sizeof(sectionName));
-		CLEO_ReadStringPointerOpcodeParam(thread, key, sizeof(key));
+		char strValue[MAX_STR_LEN]; OPCODE_READ_PARAM_STRING_BUFF(strValue, sizeof(strValue));
+		auto path = OPCODE_READ_PARAM_FILEPATH();
+		OPCODE_READ_PARAM_STRING_BUFF(section, sizeof(section));
+		OPCODE_READ_PARAM_STRING_BUFF(key, sizeof(key));
 
-		CLEO_ResolvePath(thread, path, sizeof(path)); // convert to absolute path
-
-		result = WritePrivateProfileString(sectionName, key, strValue, path);
-
-		CLEO_SetThreadCondResult(thread, result);
+		auto result = WritePrivateProfileString(section, key, strValue, path);
 
+		OPCODE_CONDITION_RESULT(result);
 		return OR_CONTINUE;
 	}
 } iniFiles;
+
+char IniFiles::section[128] = { 0 };
+char IniFiles::key[128] = { 0 };
```

Comment:
is this necessary?

Reply:
if yes, why we initialize buffers here and not where we declare them?

static char section[128] = { 0 };
static char key[128] = { 0 };

Reply:
Ok, statics are by default initialize with zeros.